src/manager.rs

```diff
@@ -120,31 +120,38 @@ impl<S: MutEventSubscriber> EventManager<S> {
     }
 
     fn dispatch_events(&mut self, event_count: usize) {
+        // Used to record whether there's an endpoint event that needs to be handled.
+        #[cfg(feature = "remote_endpoint")]
+        let mut endpoint_event = None;
+
         // Use the temporary, pre-allocated buffer to check ready events.
         for ev_index in 0..event_count {
             let event = self.ready_events[ev_index];
+            let fd = event.fd();
 
-            #[cfg(feature = "remote_endpoint")]
-            {
-                if self.dispatch_endpoint_event(event) {
-                    continue;
+            if let Some(subscriber_id) = self.epoll_context.subscriber_id(fd) {
+                self.subscribers.get_mut_unchecked(subscriber_id).process(
+                    Events::with_inner(event),
+                    // The `subscriber_id` is valid because we checked it before.
+                    &mut self.epoll_context.ops_unchecked(subscriber_id),
+                );
+            } else {
+                #[cfg(feature = "remote_endpoint")]
```

Comment:
Can you add a comment here as well to why we're dispatching the other events first?